R/evaluate_effects.R

```diff
@@ -78,21 +78,23 @@ fSquared <- function(seminr_model, iv, dv) {
 }
 
 model_fsquares <- function(seminr_model) {
-  if (any(names(seminr_model$measurement_model) == "orthogonal_interaction")
-      | any(names(seminr_model$measurement_model) == "two_stage_interaction")
-      | any(names(seminr_model$measurement_model) == "scaled_interaction" )) {
-    return("The fSquare cannot be calculated as the model contains an interaction term and omitting either the antecedent or moderator in the interaction term will cause model estimation to fail")
-  }
-
   path_matrix <- seminr_model$path_coef
   fsquared_matrix <- path_matrix
-
   for (dv in all_endogenous(seminr_model$smMatrix)) {
-    for (iv in all_exogenous(seminr_model$smMatrix)) {
+    ifelse(length(grep("\\*",seminr_model$smMatrix[seminr_model$smMatrix[,2] == dv,1]) ) > 0,
+      int_components <- unique(unlist(strsplit(seminr_model$smMatrix[seminr_model$smMatrix[,2] == dv,1][grep("\\*",seminr_model$smMatrix[seminr_model$smMatrix[,2] == dv,1])], "\\*"))),
+      int_components <- NA)
+    for (iv in setdiff(all_exogenous(seminr_model$smMatrix), int_components)) {
       fsquared_matrix[iv, dv] <- fSquared(seminr_model = seminr_model,
                                           iv = iv,
                                           dv = dv)
+      fsquared_matrix[int_components,dv] <- NA
     }
   }
+  if (any(names(seminr_model$measurement_model) == "orthogonal_interaction")
```

Comment:
isn't this if condition also just checking if interactions exist? why two ways of doing it (looking through smMatrix, and looking and measurement_model)?